Maple2.Server.Game/GameServer.cs

```diff
@@ -87,6 +93,54 @@ public GameServer(FieldManager.Factory fieldFactory, PacketRouter<GameSession> r
         return gameEvent;
     }
 
+    public Shop? FindShop(GameSession session, int shopId) {
+        using GameStorage.Request db = gameStorage.Context();
+        if (!shopCache.TryGetValue(shopId, out Shop? shop)) {
+            shop = db.GetShop(shopId);
+            if (shop != null) {
+                shopCache[shopId] = shop;
+            }
+            if (shop?.RestockTime == 0) { // everything else would be player-based shops that would get refreshed
+                if (!shopItemCache.TryGetValue(shopId, out Dictionary<int, ShopItem>? shopItems)) {
+                    shopItems = db.GetShopItems(shopId).ToDictionary(item => item.Id);
+                    shopItemCache[shopId] = shopItems;
+                }
+                foreach ((int shopItemId, ShopItem shopItem) in shopItems) {
+                    Item? item = session.Item.CreateItem(shopItem.ItemId, shopItem.Rarity, shopItem.Quantity);
```

Comment:
this causes the item to be created in db, is that necessary for shop items before they are purchased? (do they need UID) it would flood the database with useless entries.

Reply:
I don't think it does get created in db. CreateItem doesn't add it to db.

Reply:
You're right, i think i looked at the wrong function.